internal/oci/runtime_vm.go

```diff
@@ -214,9 +216,13 @@ func (r *runtimeVM) CreateContainer(ctx context.Context, c *Container, cgroupPar
 
 	createdCh := make(chan error)
 
+	// Create a context with timeout for the task creation
+	taskCtx, taskCancel := context.WithTimeout(ctx, timeout)
+	defer taskCancel()
```

Comment:
You're using both `context.WithTimeout` and `time.After` with the same timeout value. This is redundant and could cause subtle race conditions where one fires slightly before the other.

I's say we use the context timeout OR the select timeout, not both:

```
taskCtx, taskCancel := context.WithTimeout(ctx, timeout)
defer taskCancel()

// ... goroutine ...

select {
case err = <-createdCh:
    // ...
case <-taskCtx.Done():
    if err := r.remove(c.ID(), ""); err != nil {
        return fmt.Errorf("failed to cleanup container after creation timeout: %w", err)
    }
    <-createdCh  // wait for goroutine
    return fmt.Errorf("CreateContainer timeout (%v)", timeout)
}
```

Reply:
@saschagrunert, i did it a bit different, does it make sense now?